src/robotFactory.js

```diff
@@ -1,10 +1,99 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
+    if (coords.x === undefined) {
+      coords.x = 0;
+    }
 
-class FlyingRobot {}
+    if (coords.y === undefined) {
+      coords.y = 0;
+    }
 
-class DeliveryDrone {}
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+
+    return this.coords.y;
```

Comment:
```suggestion
  goForward(step = 1) {
    this.coords.y += step;
```
Movement methods shouldn't return anything